Resteasy  Reactive -  stream cannot be null

**Describe the bug**

I am right now starting a new quarkus project on 1.11.0.Final. I wanted to try out the new Resteasy reactive. However, on dev mode launch I get an exception which I can't fix myself.

If I switch to Resteasy Classic, no such exception occurs.

**Expected behavior**
Switching between Resteasy Reactive and Resteasy Classic should be as painless as possible. No random exceptions should occur because of this switch.

**Actual behavior**
```
2021-01-17 17:35:14,879 INFO  [org.jbo.threads] (main) JBoss Threads version 3.2.0.Final
2021-01-17 17:35:15,104 INFO  [org.hib.Version] (build-10) HHH000412: Hibernate ORM core version 5.4.27.Final
2021-01-17 17:35:15,187 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.BeanArchiveProcessor#build threw an exception: java.lang.IllegalArgumentException: stream cannot be null
        at org.jboss.jandex.Indexer.index(Indexer.java:1595)
        at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:83)
        at io.quarkus.arc.deployment.BeanArchiveProcessor.build(BeanArchiveProcessor.java:61)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:274)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:150)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:57)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:78)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:372)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:55)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:128)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:81)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.BeanArchiveProcessor#build threw an exception: java.lang.IllegalArgumentException: stream cannot be null
        at org.jboss.jandex.Indexer.index(Indexer.java:1595)
        at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:83)
        at io.quarkus.arc.deployment.BeanArchiveProcessor.build(BeanArchiveProcessor.java:61)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

        at io.quarkus.builder.Execution.run(Execution.java:116)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:148)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:272)
        ... 9 more
Caused by: java.lang.IllegalArgumentException: stream cannot be null
        at org.jboss.jandex.Indexer.index(Indexer.java:1595)
        at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:83)
        at io.quarkus.arc.deployment.BeanArchiveProcessor.build(BeanArchiveProcessor.java:61)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:501)

2021-01-17 17:35:15,187 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
```


**To Reproduce**

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

Or attach an archive containing the reproducer to the issue.

Steps to reproduce the behavior:
1. Start the database with docker-compose up -d
2. Download the reproducer: 
[settingsservice.zip](https://github.com/quarkusio/quarkus/files/5826448/settingsservice.zip)
3. start quarkus in dev mode: mvn clean compile quarkus:dev
4. Exception from above happens
5. Now, simply replace the quarkus-resteasy-reactive-jackson extensions with quarkus-resteasy-jackson (in the settingsservice-service modul)
6. start quarkus in dev mode again
7. This exception does not occur

**Configuration**
```properties
quarkus.http.cors=true

quarkus.datasource.db-kind=mariadb
quarkus.liquibase.migrate-at-start=true

%dev.quarkus.hibernate-orm.log.sql=false
%dev.quarkus.datasource.username=root
%dev.quarkus.datasource.password=root
%dev.quarkus.datasource.jdbc.url=jdbc:mariadb://localhost:3400/settings

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 Linux martin 5.8.0-38-generic #43~20.04.1-Ubuntu SMP Tue Jan 12 16:39:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: 
 openjdk version "11.0.7" 2020-04-14 LTS
OpenJDK Runtime Environment Zulu11.39+15-CA (build 11.0.7+10-LTS)
OpenJDK 64-Bit Server VM Zulu11.39+15-CA (build 11.0.7+10-LTS, mixed mode)
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
 1.11.0.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/martin/.sdkman/candidates/maven/current
Java version: 11.0.7, vendor: Azul Systems, Inc., runtime: /home/martin/.sdkman/candidates/java/11.0.7-zulu
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.8.0-38-generic", arch: "amd64", family: "unix"

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/simple/FieldInjectedSubClassResource.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/simple/FieldInjectedSubClassResource.java
@@ -1,5 +1,6 @@
 package io.quarkus.resteasy.reactive.server.test.simple;
 
+import javax.validation.constraints.NotNull;
 import javax.ws.rs.BeanParam;
 import javax.ws.rs.GET;
 import javax.ws.rs.HeaderParam;
@@ -45,7 +46,7 @@ public class FieldInjectedSubClassResource extends FieldInjectedResource {
     public String param(@QueryParam("query") String query,
             @HeaderParam("header") String header,
             @Context UriInfo uriInfo,
-            @BeanParam BeanParamSubClass beanParam) {
+            @BeanParam @NotNull BeanParamSubClass beanParam) {
         checkInjections("/injection-subclass/param2", query, header, uriInfo, this.beanParam, query, uriInfo, beanParam);
         return "OK";
     }

--- a/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java
+++ b/independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/scanning/ResteasyReactiveScanner.java
@@ -121,17 +121,17 @@ public class ResteasyReactiveScanner {
             // FIXME: this isn't right wrt generics
             switch (target.kind()) {
                 case FIELD:
-                    beanParams.add(target.asField().type().toString());
+                    beanParams.add(target.asField().type().name().toString());
                     break;
                 case METHOD:
                     Type setterParamType = target.asMethod().parameters().get(0);
-                    beanParams.add(setterParamType.toString());
+                    beanParams.add(setterParamType.name().toString());
                     break;
                 case METHOD_PARAMETER:
                     MethodInfo method = target.asMethodParameter().method();
                     int paramIndex = target.asMethodParameter().position();
                     Type paramType = method.parameters().get(paramIndex);
-                    beanParams.add(paramType.toString());
+                    beanParams.add(paramType.name().toString());
                     break;
                 default:
                     break;

